backend/crud.py

```diff
@@ -197,14 +265,231 @@ def update_schema(db: Session, schema_id: int, data: SchemaUpdateSchema) -> Sche
+            data[attr] = None
+    return data
+
+
+def _get_attr_values_batch(db: Session, entities: List[Entity], attrs_to_include: List[AttributeDefinition]) -> List[dict]:
+    '''Gets attr. values for list of entities by splitting attrs in
+    groups by type to select multiple attributes for all entities
+    in one query
+    '''
+    results_map = {
+        entity.id: {
+            'id': entity.id, 
+            'slug': entity.slug, 
+            'deleted': entity.deleted, 
+            'name': entity.name
+        } 
+        for entity in entities
+    }
+
+    for i in results_map.values():
+        for attr_def in attrs_to_include:
+            if attr_def.list:
+                i.update({attr_def.attribute.name: []})
+            else:
+                i.update({attr_def.attribute.name: None})
+
+    attr_groups: Dict[str, List[Attribute]] = defaultdict(list)
+    attributes = [i.attribute for i in attrs_to_include]
+    for attr in attributes:
+        attr_groups[attr.type.name].append(attr)
+    
+    ent_ids = [i.id for i in entities]
+    attr_map = {i.attribute.id: i.attribute.name for i in attrs_to_include}
+    for group, attrs in attr_groups.items():
+        value_model: Value = AttrType[group].value.model
+        q = (
+            select(value_model)
+            .where(value_model.entity_id.in_(ent_ids))
+            .where(value_model.attribute_id.in_([i.id for i in attrs]))
+        )
+        rows = db.execute(q).scalars().all()
+        for r in rows:
+            ent: dict = results_map[r.entity_id]
+            attr: str = attr_map[r.attribute_id]
+            if isinstance(ent[attr], list):
+                ent[attr].append(r.value)
+            else:
+                ent[attr] = r.value
+    
+    results = list(results_map.values())  
+    return results
+
+
+FILTER_MAP = {
+    'eq': '__eq__',
+    'lt':'__lt__',
+    'gt': '__gt__',
+    'le': '__le__',
+    'ge': '__ge__',
+    'ne': '__ne__'
+}
+
+ALLOWED_FILTERS = {
+    AttrType.STR: ['eq', 'lt', 'gt', 'le', 'ge', 'ne'],
+    AttrType.INT: ['eq', 'lt', 'gt', 'le', 'ge', 'ne'],
+    AttrType.FLOAT: ['eq', 'lt', 'gt', 'le', 'ge', 'ne'],
+    AttrType.DT: ['eq', 'lt', 'gt', 'le', 'ge', 'ne'],
+    AttrType.BOOL: ['eq', 'ne']
+}
+
+
+def _parse_filters(filters: dict, attrs: List[str]) -> Tuple[Dict[str, Dict[str, Any]], Dict[str, Dict[str, Any]]]:
+    '''Returns tuple of two `dict`s like `{attr_name: {op1: value, op2: value}}`.
+    First `dict` is for attribute filters, second is for `Entity.name` filters
+    '''
+    attrs_filters = defaultdict(dict)
+    name_filters = {}
+    for f, v in filters.items():
+        split = f.split('.')
+        attr = '.'.join(split[:-1]) if len(split) > 1 else split[0]
+        filter = 'eq' if len(split) == 1 else split[-1]
+        if attr == 'name':
+            name_filters[filter] = v
+            continue
+        elif attr not in attrs or filter not in FILTER_MAP:
+            continue
```

Comment:
It depends on whether you see passing filter with wrong attr. name/operator as a huge mistake that requires raising an exception

Reply:
I agree, it will not be a _huge mistake_. I'm only worried that it would be very difficult to debug client code, if the attribute or operator name is incorrect. So I would have been in favor of presenting an error 4xx to the client, which complains about the invalid/unknown filter

Reply:
Alright, I'll add an exception then